code/ai/aicode.cpp

```diff
@@ -16139,6 +16151,8 @@ void maybe_cheat_fire_synaptic(object *objp)
 		ship	*shipp;
 		int	wing_index, time;
 
+		Assertion(objp->type == OBJ_SHIP, "maybe_cheat_fire_synaptic was called with a non-ship object type of %d. Please report!", objp->type);
```

Comment:
IMHO it's nice to use assertion failure messages that explain the context in English rather than with function names,  given that it's user-facing text, but it's not essential. Striving for that has been helpful in my own PRs to ensure I understand what the assertion failing would really mean.

Ditto for other assertions this PR adds.

Reply:
I think I've cleared them up.